File: frontend/src/types/project.ts
export interface Project {
  id: string;
  name: string;
  description: string | null;
  latitude: number | null;
  longitude: number | null;
  elevation: number | null;
  created_at: string;
  updated_at: string;
  dataset_count: number;
}

export interface ProjectListResponse {
  projects: Project[];
  total: number;
}

export interface ProjectPayload {
  name: string;
  description?: string | null;
  latitude?: number | null;
  longitude?: number | null;
  elevation?: number | null;
}

File: backend/app/schemas/project.py
from __future__ import annotations

import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class ProjectCreate(BaseModel):
    name: str = Field(min_length=1, max_length=255)
    description: str | None = None
    latitude: float | None = Field(default=None, ge=-90, le=90)
    longitude: float | None = Field(default=None, ge=-180, le=180)
    elevation: float | None = None


class ProjectUpdate(BaseModel):
    name: str | None = Field(default=None, min_length=1, max_length=255)
    description: str | None = None
    latitude: float | None = Field(default=None, ge=-90, le=90)
    longitude: float | None = Field(default=None, ge=-180, le=180)
    elevation: float | None = None


class ProjectResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    name: str
    description: str | None
    latitude: float | None
    longitude: float | None
    elevation: float | None
    created_at: datetime
    updated_at: datetime
    dataset_count: int = 0


class ProjectListResponse(BaseModel):
    projects: list[ProjectResponse]
    total: int